File: ios/Runner/Database/Models/Group/GroupConversationModel.swift
//
//  GroupConversationModel.swift
//  Runner
//
//  Created by Hashim Khan on 16/12/2024.
//


import Foundation

struct GroupConversationModel: Codable, Equatable {
    var id: Int?
    var name: String?
    var conversations: [GroupInnerConversationModel]?
    var unreadCount: Int?
    var conversationsCount: Int?
    var groupSettings: GroupSettingsModel?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case conversations
        case unreadCount = "unread_count"
        case conversationsCount = "conversations_count"
        case groupSettings = "group_setting"
    }
}

File: android/app/src/main/kotlin/com/transport_system/ts_admin/data_providers/models/group/GroupConversationModel.kt
package com.transport_system.ts_admin.data_providers.models.group

import com.google.gson.annotations.SerializedName

data class GroupConversationModel(
    val id: Int?,
    val name: String?,
    val conversations: List<GroupInnerConversationModel>?,
    @SerializedName("unread_count") val unreadCount: Int?,
    @SerializedName("conversations_count") val conversationsCount: Int?,
    @SerializedName("group_setting") val groupSettings: GroupSettingsModel?
)
